src/main/java/com/direwolf20/buildinggadgets/common/util/tools/ToolRenders.java

```diff
@@ -404,22 +398,12 @@ private static void renderDestructionOverlay(EntityPlayer player, World world, B
         //Enable Blending (So we can have transparent effect)
         GlStateManager.enableBlend();
         //This blend function allows you to use a constant alpha, which is defined later
-        //GlStateManager.blendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
         GlStateManager.blendFuncSeparate(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA, GlStateManager.SourceFactor.ONE, GlStateManager.DestFactor.ZERO);
-        List<BlockPos> sortedCoordinates = SortingHelper.Blocks.byDistance(coordinates, player); //Sort the coords by distance to player.
 
         Tessellator t = Tessellator.getInstance();
         BufferBuilder bufferBuilder = t.getBuffer();
-        /*ArrayList<EnumFacing> directions = GadgetDestruction.assignDirections(facing, player);
-        BlockPos a = new BlockPos(0,0,0);
-        a = a.offset(directions.get(0), GadgetDestruction.getToolValue(stack, NBTKeys.GADGET_VALUE_LEFT));
-        a = a.offset(directions.get(2), GadgetDestruction.getToolValue(stack, NBTKeys.GADGET_VALUE_UP));
-        BlockPos b = new BlockPos(0,0,0);
-        b = b.offset(directions.get(1), GadgetDestruction.getToolValue(stack, NBTKeys.GADGET_VALUE_RIGHT));
-        b = b.offset(directions.get(2), GadgetDestruction.getToolValue(stack, NBTKeys.GADGET_VALUE_DOWN));
-        b = b.offset(directions.get(4), GadgetDestruction.getToolValue(stack, NBTKeys.GADGET_VALUE_DEPTH));*/
-
-        for (BlockPos coordinate : sortedCoordinates) {
+
+        for (BlockPos coordinate : GadgetDestruction.getClearingPositionsForRendering(world, startBlock, facing, player, heldItem)) {
```

Comment:
If I'm not wrong then this shouldn't be using the destruction gadget for the positions. It should be using the api. @MajorTuvok can you confirm.

Reply:
I don't know what part of the Api would be useful for that...? If you ask me all these Methods to get the rendering positions should be a Method on GadgetGeneric, but you were already woeking on that in 1.14 right?

Reply:
I'm referring to your simplicfication of ToolRenders.

Reply:
I'm moving all the rendering code to a static upon the each of the Gadgets but not for the actual selection and computation of what blocks need to be effected. For all the other tools this works though the BuildingMode from the api. I guess because the destruction gadget doesn't do that then this is likely right.

Reply:
Note: The Copy-Paste Gadget doesn't do that either...

Reply:
That's true. Then yeah, I guess doing it this way is fine.